src/App.js

```diff
@@ -74,44 +79,74 @@ export default function App() {
             moves: getMoves(moveSet),
         };
         return pokemonObj;
-    }
+    }, []);
 
     useEffect(() => {
-        fetch(`https://pokeapi.co/api/v2/pokemon/${randomNumber(251)}`)
-            .then((response) => response.json())
-            .then((response) => {
-                let pokemonObj = generatePokemon(response, "back");
-                setplayerPokemonObject(pokemonObj);
-                setPlayerHP(pokemonObj.hp);
-            });
-
-        fetch(`https://pokeapi.co/api/v2/pokemon/${randomNumber(251)}`)
-            .then((response) => response.json())
-            .then((response) => {
-                let pokemonObj = generatePokemon(response, "front");
-
-                setOppPokemonObject(pokemonObj);
-                setOpponentHP(pokemonObj.hp);
-            });
-    }, []);
+        if (playerPokemonObject.name === "") {
```

Comment:
you could maybe wrap each fetch in a check for each pokemonObj.name being empty string. But that's not a huge deal